synapseclient/models/protocols/activity_protocol.py

```diff
@@ -0,0 +1,90 @@
+"""Protocol for the specific methods of this class that have synchronous counterparts
+generatred at runtime."""
+
+from typing import Optional, TYPE_CHECKING, Protocol, Union
+
+from synapseclient import Synapse
+
+if TYPE_CHECKING:
+    from synapseclient.models import File, Table, Activity
+
+
+class ActivitySynchronousProtocol(Protocol):
+    """
+    The protocol for methods that are asynchronous but also
+    have a synchronous counterpart that may also be called.
+    """
+
+    def store(
+        self,
+        parent: Optional[Union["Table", "File"]] = None,
```

Comment:
should we not have it default to None here? have it a required arg? would storing the activity without a parent lead to an error

Reply:
Not in this case. I still need to add some input validation (SYNPY-1400), but Activity can exist without a parent. Activity is it's concept where other Synapse entities (File, Table, ect...) have inbound edges pointing at the activity:

![image](https://github.com/Sage-Bionetworks/synapsePythonClient/assets/17128019/c96837cd-3eb1-4080-9009-47a535192a76)


This means that the Activity has its own ID that can be used to store the activity with and update information.

Reply:
Interesting, okay. Thanks for the diagram!